apps/managedidentity/managedidentity.go

```diff
@@ -204,28 +398,76 @@ func (client Client) getTokenForRequest(req *http.Request) (accesstokens.TokenRe
 			Err:  fmt.Errorf("http call(%s)(%s) error: reply status code was %d", req.URL.String(), req.Method, resp.StatusCode),
 		}
 	}
-	var r accesstokens.TokenResponse
+
 	err = json.Unmarshal(responseBytes, &r)
 	return r, err
 }
 
-// Acquires tokens from the configured managed identity on an azure resource.
-//
-// Resource: scopes application is requesting access to
-// Options: [WithClaims]
-func (client Client) AcquireToken(ctx context.Context, resource string, options ...AcquireTokenOption) (base.AuthResult, error) {
-	o := AcquireTokenOptions{}
+func (c *Client) handleAzureArcResponse(response *http.Response, ctx context.Context, claims string, platform string) (accesstokens.TokenResponse, error) {
+	if response.StatusCode == http.StatusUnauthorized {
+		wwwAuthenticateHeader := response.Header.Get(wwwAuthenticateHeaderName)
 
-	for _, option := range options {
-		option(&o)
-	}
-	req, err := createIMDSAuthRequest(ctx, client.miType, resource, o.claims)
-	if err != nil {
-		return base.AuthResult{}, err
-	}
-	tokenResponse, err := client.getTokenForRequest(req)
-	if err != nil {
-		return base.AuthResult{}, err
+		if len(wwwAuthenticateHeader) == 0 {
+			return accesstokens.TokenResponse{}, errors.New("response has no www-authenticate header")
+		}
+
+		var secretFilePath string
```

Comment:
nit: it will be more readable to have the file path validation in another method.